detectorSegmentations/src/GridDRcalo_k4geo.cpp

```diff
@@ -68,13 +70,64 @@ namespace DDSegmentation {
     auto rot = paramBase->GetRotationZYX(noPhi);
     auto translation = rot * localPos;
 
+    // we want the cell position to be the front end of the tower
+    // retrieve the fiber length
+    int row = y(cID);
+    int col = x(cID);
+    double fiberLen = paramBase->GetShortFibers(noEta).retrieveFiberLength(row, col);
+    // and vector to the front end
+    auto diff = paramBase->GetTowerPos(noPhi) - waferPos;
+    auto unitVec = diff / std::sqrt(diff.Mag2());
+
+    // total vector is sum of the vector to the wafer center + rotated local coordinate
+    // + the translation to the front end of the tower
+    auto total = translation + waferPos + unitVec * fiberLen;
+
+    // if LHS rotate by 180 deg w.r.t. X axis (on par to the DRconstructor)
```

Comment:
If not there already, can you add a comment in the DRConstructor, where you do the rotation, that downstream code rely on this i.e. if for some reason we change this in the future, the person doing so will know that more modifications are needed?